src/lib/configs/project/dockerCompose.ts

```diff
@@ -0,0 +1,315 @@
+      items: { type: "string" },
+      nullable: true,
+    },
+    secrets: {
+      type: "object",
+      nullable: true,
+      additionalProperties: {
+        type: "object",
+        properties: {
+          file: { type: "string", nullable: true },
+        },
+        required: [],
+      },
+      required: [],
+    },
+  },
+  required: ["version", "services"],
+};
+
+const NOX_IPFS_MULTIADDR = `/dns4/${IPFS_CONTAINER_NAME}/tcp/${IPFS_PORT}`;
+
+type GenNoxImageArgs = {
+  name: string;
+  tcpPort: number;
+  webSocketPort: number;
+  bootStrapNoxName: string;
+};
+
+function genNox({
+  name,
+  tcpPort,
+  webSocketPort,
+  bootStrapNoxName,
+}: GenNoxImageArgs): [name: string, service: Service] {
+  return [
+    name,
+    {
+      image: versions.nox,
+      pull_policy: "always",
+      ports: [`${tcpPort}:${tcpPort}`, `${webSocketPort}:${webSocketPort}`],
+      environment: {
+        FLUENCE_ENV_AQUA_IPFS_EXTERNAL_API_MULTIADDR,
+        FLUENCE_ENV_DECIDER_IPFS_MULTIADDR: NOX_IPFS_MULTIADDR,
+        FLUENCE_ENV_AQUA_IPFS_LOCAL_API_MULTIADDR: NOX_IPFS_MULTIADDR,
+        FLUENCE_ENV_CONNECTOR_API_ENDPOINT: `http://${CHAIN_CONTAINER_NAME}:${CHAIN_PORT}`,
+        FLUENCE_ENV_CONNECTOR_FROM_BLOCK: "earliest",
+        WASM_LOG: "info",
+        RUST_LOG:
+          "debug,particle_reap=debug,aquamarine=warn,aquamarine::particle_functions=debug,aquamarine::log=debug,aquamarine::aqua_runtime=error,ipfs_effector=off,ipfs_pure=off,system_services=debug,marine_core::module::marine_module=info,tokio_threadpool=info,tokio_reactor=info,mio=info,tokio_io=info,soketto=info,yamux=info,multistream_select=info,libp2p_secio=info,libp2p_websocket::framed=info,libp2p_ping=info,libp2p_core::upgrade::apply=info,libp2p_kad::kbucket=info,cranelift_codegen=info,wasmer_wasi=info,cranelift_codegen=info,wasmer_wasi=info,run-console=trace,wasmtime_cranelift=off,wasmtime_jit=off,libp2p_tcp=off,libp2p_swarm=off,particle_protocol::libp2p_protocol::upgrade=info,libp2p_mplex=off,particle_reap=off,netlink_proto=warn",
+        FLUENCE_SYSTEM_SERVICES__ENABLE: "aqua-ipfs,decider",
+        FLUENCE_ENV_CONNECTOR_WALLET_KEY:
+          "0xfdc4ba94809c7930fe4676b7d845cbf8fa5c1beae8744d959530e5073004cf3f",
+        FLUENCE_ENV_CONNECTOR_CONTRACT_ADDRESS:
+          "0x0f68c702dC151D07038fA40ab3Ed1f9b8BAC2981",
+        FLUENCE_SYSTEM_SERVICES__DECIDER__DECIDER_PERIOD_SEC: 10,
+        FLUENCE_MAX_SPELL_PARTICLE_TTL: "9s",
+        FLUENCE_SYSTEM_SERVICES__DECIDER__NETWORK_ID: 31337,
+        FLUENCE_ROOT_KEY_PAIR__PATH: `/run/secrets/${name}`,
+      },
+      command: [
+        "--aqua-pool-size=5",
+        `-t=${tcpPort}`,
+        `-w=${webSocketPort}`,
+        "--external-maddrs",
+        `/dns4/${name}/tcp/${tcpPort}`,
+        `/dns4/${name}/tcp/${webSocketPort}/ws`,
+        "--allow-private-ips",
+        tcpPort === TCP_PORT_START
+          ? "--local"
+          : `--bootstraps=/dns/${bootStrapNoxName}/tcp/${TCP_PORT_START}`,
```

Comment:
Why some things are through ENV variables, and others are passed by flags?

I think it makes sense to pass as much information as possible via `--` flags. And everything else – via `Config.toml`

Reply:
It's important to move as much as possible to Config.toml because Config.toml will be used in different networks-environments, while docker-compose is only for local.

So moving to Config.toml makes it solved for both public networks, and for local.

Reply:
I am currently just doing the same stuff I hade in docker-compose in cli repo - I will imporve and extend all this later
You can suggest which porperties can be passed how better - would be useful

Reply:
I am currently not generating Config.toml at all. I will move everything possible there in the next step